server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -0,0 +1,424 @@
+                    String message = "Processor type [" + processor.getType() + "] is not installed on node [" + entry.getKey() + "]";
+                    exceptions.add(ConfigurationUtils.newConfigurationException(processor.getType(), processor.getTag(), null, message));
+                }
+            }
+        }
+        ExceptionsHelper.rethrowAndSuppress(exceptions);
+    }
+
+    public void deletePipeline(DeleteSearchPipelineRequest request, ActionListener<AcknowledgedResponse> listener) throws Exception {
+        clusterService.submitStateUpdateTask(
+            "delete-search-pipeline-" + request.getId(),
+            new AckedClusterStateUpdateTask<>(request, listener) {
+                @Override
+                public ClusterState execute(ClusterState currentState) {
+                    return innerDelete(request, currentState);
+                }
+
+                @Override
+                public ClusterManagerTaskThrottler.ThrottlingKey getClusterManagerThrottlingKey() {
+                    return deletePipelineTaskKey;
+                }
+
+                @Override
+                protected AcknowledgedResponse newResponse(boolean acknowledged) {
+                    return new AcknowledgedResponse(acknowledged);
+                }
+            }
+        );
+
+    }
+
+    static ClusterState innerDelete(DeleteSearchPipelineRequest request, ClusterState currentState) {
+        SearchPipelineMetadata currentMetadata = currentState.metadata().custom(SearchPipelineMetadata.TYPE);
+        if (currentMetadata == null) {
+            return currentState;
+        }
+        Map<String, PipelineConfiguration> pipelines = currentMetadata.getPipelines();
+        Set<String> toRemove = new HashSet<>();
+        for (String pipelineKey : pipelines.keySet()) {
+            if (Regex.simpleMatch(request.getId(), pipelineKey)) {
+                toRemove.add(pipelineKey);
+            }
+        }
+        if (toRemove.isEmpty()) {
+            if (Regex.isMatchAllPattern(request.getId())) {
+                // Deleting all the empty state is a no-op.
+                return currentState;
+            }
+            throw new ResourceNotFoundException("pipeline [{}] is missing", request.getId());
+        }
+        final Map<String, PipelineConfiguration> newPipelines = new HashMap<>(pipelines);
+        for (String key : toRemove) {
+            newPipelines.remove(key);
+        }
+        ClusterState.Builder newState = ClusterState.builder(currentState);
+        newState.metadata(
+            Metadata.builder(currentState.getMetadata()).putCustom(SearchPipelineMetadata.TYPE, new SearchPipelineMetadata(newPipelines))
+        );
+        return newState.build();
+    }
+
+    public SearchRequest transformRequest(SearchRequest originalRequest) {
+        String pipelineId = originalRequest.pipeline();
+        if (pipelineId != null) {
+            PipelineHolder pipeline = pipelines.get(pipelineId);
```

Comment:
Non-blocking nit - if we want to keep the name of `PipelineHolder `, rename `pipeline` to `pipelineholder` to avoid invoke  `pipeline.pipeline`